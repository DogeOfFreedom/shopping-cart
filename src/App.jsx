import { useParams } from "react-router-dom";
import NavBar from "./navbar/NavBar";
import "./index.css";
import Home from "./pages/Home";
import Shop from "./pages/Shop";
import { createContext, useState } from "react";
import Checkout from "./checkout/Checkout";

export const ThemeContext = createContext({
  theme: "dark",
});

function App() {
  const { name } = useParams();
  const duration = 150;
  const [fadeOut, setFadeOut] = useState(false);
  const [cart, setCart] = useState([]);
  const [checkout, setCheckout] = useState(false);
  const [theme, setTheme] = useState("dark");

  const showCheckout = () => {
    cart.length && setCheckout(true);
  };
  const closeCheckout = () => setCheckout(false);

  const countItems = () => {
    let count = 0;
    cart.forEach((item) => {
      count += item.quantity;
    });
    return count;
  };

  return (
    <ThemeContext.Provider value={{ theme, setTheme }}>
      <div>
        {checkout && (
          <Checkout show={checkout} closeCheckout={closeCheckout} cart={cart} />
        )}
        <NavBar
          items={countItems()}
          duration={duration}
          setFadeOut={setFadeOut}
          showCheckout={showCheckout}
        />
        {name === "home" ? (
          <Home duration={duration} fadeOut={fadeOut} setFadeOut={setFadeOut} />
        ) : (
          <Shop
            fadeOut={fadeOut}
            setFadeOut={setFadeOut}
            cart={cart}
            addToCart={setCart}
          />
        )}
      </div>
    </ThemeContext.Provider>
  );
}

export default App;
