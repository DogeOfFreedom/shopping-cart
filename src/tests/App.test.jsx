import { render, screen } from "@testing-library/react";
import App from "../App";
import { describe, expect, it, vi } from "vitest";
import { MemoryRouter } from "react-router-dom";

const mocks = vi.hoisted(() => {
  return {
    export: vi.fn(),
  };
});

vi.mock("react-router-dom", async (importOriginal) => {
  const actual = await importOriginal();
  return {
    ...actual,
    useParams: mocks.export,
  };
});

describe("App Component", () => {
  it("renders home component on '/home' route", () => {
    mocks.export.mockReturnValue({ name: "home" });

    render(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    );

    expect(screen.queryByText(/Menzei's got you/i)).toBeInTheDocument();
  });

  it("renders shop component on '/shop' route", async () => {
    mocks.export.mockReturnValue({ name: "shop" });

    render(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    );

    expect(screen.queryByTestId("loader")).toBeInTheDocument();
  });
});
