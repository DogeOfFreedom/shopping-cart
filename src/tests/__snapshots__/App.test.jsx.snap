// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App Component > renders home component by default 1`] = `
<div>
  <div>
    <nav
      class="_nav_344b47"
    >
      <div
        class="_logo_344b47"
      >
        <a
          class=""
          href="/home"
        >
          Menzei
        </a>
      </div>
      <div
        class="_navOptionsContainer_344b47"
      >
        <button
          class="_clickableTab_344b47 _selectedTab_344b47"
        >
          <span>
            Home
          </span>
        </button>
        <button
          class="_clickableTab_344b47"
        >
          <span>
            Shop
          </span>
        </button>
      </div>
      <div
        class="_checkout_344b47"
      />
    </nav>
    <h1>
      home
    </h1>
    <div
      class="_pageContainer_c151ab"
    >
      <div
        class="_contentContainer_c151ab _contentContainer_83b5c1 _fadeable_c151ab 
          "
      >
        <h3>
          Ever wanted to look like an NPC?
        </h3>
        <h1
          class="_heroText_83b5c1"
        >
          Menzei's Got You
        </h1>
        <button
          class="_shopButton_83b5c1"
        >
          Shop Now
        </button>
      </div>
    </div>
  </div>
</div>
`;
